packages/node_modules/nav-frontend-stegindikator/src/stegindikator.tsx

```diff
@@ -0,0 +1,211 @@
+import * as React from 'react';
+import * as cn from 'classnames';
+
+import { omit } from 'nav-frontend-js-utils';
+
+import 'nav-frontend-stegindikator-style';
+
+const cls = (state) => cn('stegindikator', {
+    'stegindikator--kompakt': state.kompakt
+});
+
+const stegCls = (props) => cn('stegindikator__steg', {
+    'stegindikator__steg--labelled': props.visLabel
+});
+
+const innerCls = (props) => cn('stegindikator__steg-inner', {
+    'stegindikator__steg-inner--aktiv': props.aktiv,
+    'stegindikator__steg-inner--ferdig': props.ferdig,
+    'stegindikator__steg-inner--disabled': props.disabled
+});
+
+export interface StegindikatorStegProps {
+    label: string;
+    index?: number;
+    visLabel?: boolean;
+    onClick?: () => void;
+    aktiv?: boolean;
+    ferdig?: boolean;
+    disabled?: boolean;
+}
+
+export interface NavFrontendStegindikatorProps {
+    steg: StegindikatorStegProps[];
+    visLabel: boolean;
+    kompakt: boolean;
+    aktivtSteg: number;
+    ferdigSteg: number[];
```

Comment:
Er det behov for å angi både aktivt steg og ferdige steg her? Tenker det bør være gitt hvilke steg som er ferdige utifra aktivt steg.

Reply:
Bruker kan potensielt navigere bakover blant ferdige steg, uten å nødvendigvis gjøre de ferdige stegene ugyldig/uferdig av den grunn. 

**Eksempel:** 
Bruker har gjort ferdig steg 1 og 2, og er på steg 3, men velger å klikke på 1 igjen for å se hva som stod der: 
<img width="688" alt="screen shot 2018-02-26 at 15 42 29" src="https://user-images.githubusercontent.com/74510/36676368-cd39fa52-1b0b-11e8-958b-9fdfb17ea3ab.png">

Reply:
Vil det si at man alltid må holde rede på ferdige steg i tillegg til aktivt steg @Lillebo?

Reply:
Ja, hvis man vil at ferdige steg skal markeres som ferdig da.

Reply:
Kan eventuelt legge til slik at hvis `ferdigSteg` ikke er angitt, så vil alle steg med lavere index enn `aktivtSteg` markeres som ferdig?

Reply:
Ja, det burde vi isåfall 👍 dette er egentlig et veldig special case imo, og jeg er egentlig litt usikker på om det er noe vi skal ha support for outofthebox i komponenten, ettersom man i de fleste tilfeller antagelig vil endre på innholdet i fremtidige steg gitt A eller B i et tidligere steg. Men om det fins et behov for dette så ser jeg ikke et stort problem sålenge det fungerer uten å måtte holde styr på ferdige steg til enhver tid, sånn som du foreslår.

Reply:
Hm, litt usikker på hva du mener. Tenker at default funksjonalitet bør legge til rette for at brukere fritt kan navigere mellom stegene (såfremst disse ikke er disabled), i tilfellet de glemte hva de valgte, eller hva som stod på et tidligere steg. Om bruker så foretar seg noe nytt på tidligere ferdige steg, bør ramme-applikasjonen ta seg av det å oppdatere påfølgende steg deretter (evt. gjøre senere dem ugyldige/uferdige/nullstilte). Men hvis bruker ikke foretar seg noe på tidligere steg, bør ikke de andre stegene påvirkes av det.

Eneste jeg er litt bekymret for hvis vi legger til automatisk markering av ferdige steg basert på `aktivtSteg` er at `ferdigSteg` ikke blir brukt slik det skal, og brukere som navigerer seg tilbake i stegene kan bli forvirret når et steg som var markert som ferdig plutselig ikke ser ut som det lenger.

Reply:
Kunne det vært en idé å splitte funksjonaliteten opp i to komponenter da? En som er den vanlige stegindikatoren som viser steg som ferdige/uferdige basert på aktivt steg, og en mer interaktiv stegindikator-komponent som benytter seg av førstnevnte, men legger på kompleksiteten med å angi ferdige steg?

Reply:
Hm, vi kan muligens gjøre det skillet basert på om man angir noen `onClick` prop på steg-objektene eller ikke. Ingen `onClick` og ingen `ferdigSteg` så markeres ferdig steg automatisk basert på `aktivSteg`, hvis `onClick` er angitt så krever vi at `ferdigSteg` blir gitt? Hvis det ikke blir alt for forvirrende?

Reply:
Nja, da tenker jeg kanskje APIet blir litt forvirrende/rart? Hva skjer da om du har onClick på noen on onClick på andre? Jeg tenker det bør være veldig tydelig om man har å gjøre med en interaktiv stegindikator eller ikke når man koder det opp, i form av å si f.eks. `<StegIndikatorInteraktiv />` og `<StegIndikator />`. Alternativt gjennom en `interaktiv`-prop, men jeg tenker at førstnevnte bør være å foretrekke om det lar seg gjøre.

Reply:
Ja, kan fort bli litt rotete. Har ikke så veldig lyst til å dele den opp i flere komponenter. Da foretrekker jeg å bare gjøre som beskrevet litt lengre opp: å automatisk markere alle steg med lavere index enn `aktivtSteg` som ferdig sålenge `ferdigSteg` ikke er angitt.